google/web-stories-wp
# Taxonomies: Hierarchical input accessibility updates

## Context

Bug bash discovered that the hierarchical input needed some accessibility updates.

## Summary

Accessibility updates for the hierarchical input.

## Relevant Technical Choices

- The `listbox` role doesn't allow for nested focusable options, so the [`tree`](https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView) role works better here.

## To-do

n/a

## User-facing changes

Small visual change: the border will be highlighted when any checkbox is focused

<img src="https://user-images.githubusercontent.com/22185279/137180770-10be6704-10af-4a44-9e51-b7bd2334246c.gif" height="300px" />

## Testing Instructions






This PR can be tested by following these steps:

1. Hitting `Tab` will no longer go through the options, instead it will move away from the checkbox list
2. Tabbing into the checkbox area from the search box will do one of two things:
    - If there is a selected category then it will be focused
    - If there is not a selected category, then the first category will be focused
3. Tab into the checkbox list, change the focus with the arrow keys. Then tab back out and in. The checkbox that was focused before tabbing out should still receive focus.
4. Should be usable with a screen reader


## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #9274
